### PR TITLE
Enhance 3.x torch algorithm entry

### PR DESCRIPTION
**Pull request by yuwenzho**
## Type of Change

feature
API changed or not: no

## Description

Add new function `preprocess_quantizer` and `postprocess_model` to reduce duplicate code and fix bug in mixed algorithms.

For example:
```diff
- if getattr(model, "quantizer", False):
-     quantizer = model.quantizer
- else:
-     quantizer = RTNQuantizer(quant_config=weight_config)
+ quantizer = preprocess_quantizer(model, quantizer_cls=RTNQuantizer, quant_config=weight_config)

model = quantizer.execute(model, mode=mode)

- if getattr(model, "quantizer", False):
-     del model.quantizer
- else:
-     model.quantizer = quantizer
+ postprocess_model(model, mode, quantizer)
```

## How has this PR been tested?

CI

## Dependency Change?

No

### CODE REVIEW

File: neural_compressor/torch/algorithms/weight_only/autoround.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 class AutoRoundQuantizer(Quantizer):
     def __init__(
         self,
-        weight_config: dict = {},
+        quant_config: dict = {},
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/weight_only/autoround.py:29]: Empty dictionary is a dangerous default value https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

File: neural_compressor/torch/utils/utility.py [hunk 2]
```diff
@@ -131,3 +131,39 @@ class Mode(Enum):
...
+
+def postprocess_model(model, mode, quantizer):
+    """Process `quantizer` attribute of model according to current mode.
+
+    Args:
+        model (torch.nn.Module): pytorch model.
+        mode (Mode): The mode of current phase, including 'prepare', 'convert' and 'quantize'.
+        quantizer (Quantizer): quantizer object.
+    """
+    if mode == Mode.PREPARE:
```
Comment (yiliu30) [inline neural_compressor/torch/utils/utility.py:165]: Please add some comments to make the logic more clear.

**Review by yiliu30 — APPROVED** [review]

File: neural_compressor/torch/utils/utility.py [hunk 3]
```diff
@@ -131,3 +131,39 @@ class Mode(Enum):
     PREPARE = "prepare"
     CONVERT = "convert"
     QUANTIZE = "quantize"
+
+
+def preprocess_quantizer(model, quantizer_cls, quant_config=None, *args, **kwargs):
```
Comment (xin3he) [inline neural_compressor/torch/utils/utility.py:136]: I think `get_quantizer` or `init_quantizer` is more suitable.

**Review by xin3he — APPROVED** [review]

File: neural_compressor/torch/algorithms/weight_only/autoround.py (hunk 1 above)
Comment (yuwenzho) [reply neural_compressor/torch/algorithms/weight_only/autoround.py]: Set default value of `quant_config` here to None and pass an empty dictionary to AutoRoundProcessor if `quant_config` is None.

File: neural_compressor/torch/utils/utility.py (hunk 2 above)
Comment (yuwenzho) [reply neural_compressor/torch/utils/utility.py:170]: Add more comments in [cc79eef](https://github.com/intel/neural-compressor/pull/1779/commits/cc79eeffa6c03454c3afba66a11303f1b597f7be)

**Pull request merged**